src/crc64.c

```diff
@@ -126,33 +143,225 @@ uint64_t crc64(uint64_t crc, const unsigned char *s, uint64_t l) {
+                    "do eiusmod tempor incididunt ut labore et dolore magna "
+                    "aliqua. Ut enim ad minim veniam, quis nostrud exercitation "
+                    "ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis "
+                    "aute irure dolor in reprehenderit in voluptate velit esse "
+                    "cillum dolore eu fugiat nulla pariatur. Excepteur sint "
+                    "occaecat cupidatat non proident, sunt in culpa qui officia "
+                    "deserunt mollit anim id est laborum.";
+        printf("[calcula]: c7794709e69683b3 == %016" PRIx64 "\n",
+            (uint64_t)_crc64(0, li, sizeof(li)));
+        printf("[64speed]: c7794709e69683b3 == %016" PRIx64 "\n",
+            (uint64_t)crc64(0, (unsigned char*)li, sizeof(li)));
+        return 0;
+
+    }
+
+    int init_this_loop = 1;
+    long long init_start, init_end, mismatched = 0;
+
+    do {
+        unsigned char* data = NULL;
+        uint64_t passes = 0;
+        if (crc64_test_size) {
+            data = zmalloc(crc64_test_size);
+            if (!data) {
+                printf(
+                    "Could not allocate sufficient data to test crc64 %" PRIu64 "\n",
+                    crc64_test_size);
+                return 1;
+            }
+            genBenchmarkRandomData((char*)data, crc64_test_size);
+            // We want to hash about 1 gig of data in total, looped, to get a good
+            // idea of our performance.
+            passes = (UINT64_C(0x100000000) / crc64_test_size);
+            passes = passes >= 2 ? passes : 2;
+            passes = passes <= 1000 ? passes : 1000;
+        }
+
+        crc64_init();
+        // warm up the cache
+        set_crc64_cutoffs(crc64_test_size+1, crc64_test_size+1);
+        uint64_t expect = crc64(0, data, crc64_test_size);
+
+        if (!combine && crc64_test_size) {
+            if (csv && init_this_loop) printf("algorithm,buffer,performance,crc64_matches\n");
+
+            // get the single-character version for single-byte Redis behavior
+            set_crc64_cutoffs(0, crc64_test_size+1);
+            if (bench_crc64(data, crc64_test_size, passes, expect, "crc_1byte", csv)) return 1;
+
+            set_crc64_cutoffs(crc64_test_size+1, crc64_test_size+1);
+            // run with 8-byte "single" path, crcfaster
+            if (bench_crc64(data, crc64_test_size, passes, expect, "crcspeed", csv)) return 1;
+
+            // run with dual 8-byte paths
+            set_crc64_cutoffs(1, crc64_test_size+1);
+            if (bench_crc64(data, crc64_test_size, passes, expect, "crcdual", csv)) return 1;
+
+            // run with tri 8-byte paths
+            set_crc64_cutoffs(1, 1);
+            if (bench_crc64(data, crc64_test_size, passes, expect, "crctri", csv)) return 1;
+
+            // Be free memory region, be free.
+            zfree(data);
+            data = NULL;
+        }
+
+#define INIT_SIZE UINT64_C(0xffffffffffffffff)
```

Comment:
Still not the biggest fan of the defines that are undefined later to prevent pollution. These can just be local constants?

Reply:
Done.